feat: labels (only for assets)

BugGambit opened the pull request:
Need to add integration tests for asset.labels when labels is available in API v1.

BugGambit committed:
feat: labels (only for assets)

vincent-cognite commented on src/types/types.ts:
```diff
@@ -543,6 +548,71 @@ export interface DataSetChangeByExternalId extends DataSetPatch, ExternalId {}
+  /**
+   * Name of the label.
+   */
+  name: string;
```
no ? makes it required, right?

hhenriques replied:
`name` is required on the API right?

vincent-cognite replied:
yes (I'm just typescript curious)

polomani commented on src/types/types.ts:
```diff
@@ -543,6 +548,71 @@ export interface DataSetChangeByExternalId extends DataSetPatch, ExternalId {}
 
 export interface DataSetChangeById extends DataSetPatch, InternalId {}
 
+export type Label = ExternalId;
```
do we need this alias?

BugGambit replied:
Yes, for the UX

polomani commented on src/types/types.ts:
```diff
@@ -543,6 +548,71 @@ export interface DataSetChangeByExternalId extends DataSetPatch, ExternalId {}
+export interface LabelDefinitionFilterRequest
+  extends LabelDefinitionFilter,
+    Cursor,
+    Limit {}
```
instead of `Cursor, Limit`
extends `FilterQuery`

polomani replied:
maybe make it similar to `DataSetFilterRequest` so we don't empty types?
I mean move `filter?: LabelDefinitionFilter ` in here and unwrap the current `LabelDefinitionFilter`

polomani reviewed (commented):
looks good but I am not sure about having `blablaDefinition` everywhere
I would remove alias, use `ExternalId` type and name types as `blavlaLabel` without `Definition`

BugGambit committed:
refactor: more intuitive interfaces

polomani committed:
refactor: even more intuitive interfaces

polomani reviewed (approved):
don't merge before API release
remember to squash

BugGambit committed:
feat(events): unique values aggregate (#377)

semantic-release-bot committed:
chore(release): 2.31.0 [skip ci]

# [2.31.0](https://github.com/cognitedata/cognite-sdk-js/compare/v2.30.0...v2.31.0) (2020-06-23)

### Features

* **events:** unique values aggregate ([#377](https://github.com/cognitedata/cognite-sdk-js/issues/377)) ([d47152c](https://github.com/cognitedata/cognite-sdk-js/commit/d47152c4873274648e9bf525737810ef14e9572c))

BugGambit committed:
test: added tests for asset.label

BugGambit committed:
Merge branch 'v1' into f1/labels
